assets/js/babyYoda.js

```diff
@@ -0,0 +1,47 @@
+
+const spawnBabyYoda = (randPosX, randPosY, colour) => {
+    const babyYoda = add([
+      rect(40, 40), // placeholder until we have a sprite
+      pos(randPosX, randPosY),
+      color(colour), // colour of the box until we have a sprite
+      "babyYoda", // tag to reference later on
+    ]);
+
+        //check input on each frame
+        babyYoda.onUpdate(() => {
+
+            // Define the keyboard keys for movement
+            const MOVE_LEFT_KEY = "left";
+            const MOVE_RIGHT_KEY = "right";
+            const MOVE_UP_KEY = "up";
+            const MOVE_DOWN_KEY = "down";
+
+            const movementSpeed = 1;
+
+            // Handle keyboard input
+            onKeyDown(MOVE_LEFT_KEY, () => {
+                if (babyYoda.pos.x > 0) {
+                    babyYoda.move(-movementSpeed, 0);
+
+                }
+                
+            });
+
+            onKeyDown(MOVE_RIGHT_KEY, () => {
+                // if (babyYoda.pos.y > 0) {
+                    babyYoda.move(movementSpeed, 0);
+                // }
+ 
+            });
+
+            onKeyDown(MOVE_UP_KEY, () => {
+                babyYoda.move(0, -movementSpeed);
+            });
+
+            onKeyDown(MOVE_DOWN_KEY, () => {
+                babyYoda.move(0, movementSpeed);
+            });
```

Comment:
The conditions to prevent movement could also be added here, `0` for the up one and `height()` for the down one.